### ISSUE TITLE
Return types cleanup

### ISSUE THREAD

**Issue by jensdev** [ExtraHorizon/javascript-sdk#273]
At the moment we have constructs like
```ts
export interface Document<CustomData = null> {
  id?: ObjectId;
  userIds?: ObjectId[];
  groupIds?: ObjectId[];
  status?: string;
  data?: CustomData extends null ? Record<string, unknown> : CustomData;
  transitionLock?: {
    timestamp?: Date;
  };
  commentCount?: number;
  updateTimestamp?: Date;
  creationTimestamp?: Date;
  statusChangedTimestamp?: Date;
  creatorId?: ObjectId;
}
```

Which forces the developer to struggle with the optionals. 
```ts
const result = await sdk.data.documents.create<InvitationData>(
  SchemaIDs.INVITATIONCODES,
  { code: hash }
);

const data = result.data;

const { groupId, email: inviteEmail, userId } = data as InvitationData;
```

If the call is succesful and a document is returned. It will always have certain properties, right?

**Comment by ThomasReyskens** [comment]
@ThomasReyskens  Check server side typings

**Comment by FilipAben** [comment]
Fixed by #587. If there are other instances of this, create a new ticket. Otherwise this will be a tough one to close
